calculator.go

```diff
@@ -1,13 +1,92 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"fmt"
+	"go/token"
+	"go/types"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(a, b float64, extra ...float64) float64 {
+	var total float64 = a + b
+	for _, n := range extra {
+		total += n
+	}
+	return total
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b float64, extra []float64) float64 {
```

Comment:
We want:

```suggestion
func Subtract(a, b float64, extra ...float64) float64 {
```

because otherwise the compiler will insist on a third argument being passed, even if it's `nil`. A slice can't be optional, whereas a variadic argument (`...`) can be.